File: src/main/java/com/citasmedicas/appcitasmedicas/Service/SpecialtyService.java
package com.citasmedicas.appcitasmedicas.Service;

import com.citasmedicas.appcitasmedicas.dto.Request.CreateSpecialtyRequest;
import com.citasmedicas.appcitasmedicas.dto.Response.SpecialtyResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface SpecialtyService {
    SpecialtyResponse create(CreateSpecialtyRequest request);
    Page<SpecialtyResponse> findAll(Pageable page);
    SpecialtyResponse findById(Long id);
    void delete(Long id);
}

File: frontend/src/services/specialtyService.js
import api from './api';

export const specialtyService = {
  getAll: (page = 0, size = 100) =>
    api.get(`/specialties?page=${page}&size=${size}`),

  create: (data) =>
    api.post('/specialties', data),

  delete: (id) =>
    api.delete(`/specialties/${id}`),
};